helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvscWriter.java

```diff
@@ -173,20 +249,63 @@ protected void toJson(
                 } else {
                     gen.writeStartObject();
                     gen.writeStringField("type", schema.getName());
-                    writeProps(schema, gen);
+                    Set<String> claimedProps = executeAvscWriterPluginsForSchema(schema, gen);
+
+                    if(retainNonClaimedProps) {
+                        writeProps(schema, gen, claimedProps);
+                    }
                     gen.writeEndObject();
                 }
         }
     }
 
-    protected void aliasesToJson(Schema schema, List<AvroName> extraAliases, G gen) throws IOException {
-        Set<String> userDefinedAliases = schema.getAliases();
-        Set<String> allAliases = userDefinedAliases; //could be null
-        if (addAliasesForAvro702 && extraAliases != null) {
-            allAliases = new HashSet<>();
-            if (userDefinedAliases != null) {
-                allAliases.addAll(userDefinedAliases);
+    /***
+     * Runs execute(Schema, JsonGeneratorWrapper) for provided AvscWriterPlugins
+     * @param schema
+     * @param gen
+     * @return list of Json prop names handled by plugins
+     * @throws UnsupportedOperationException of 2+ plugins provided for same json prop
+     */
+    private Set<String> executeAvscWriterPluginsForSchema(Schema schema, G gen) {
+        Set<String> claimedProps = new HashSet<>();
+        for(AvscWriterPlugin plugin : _plugins) {
```

Comment:
you should probably have a check for plugins == null || plugins.isEmpty() and return emptySet()

Reply:
Plugins won't be null since we check this in the constructor
this._plugins = plugins == null ? Collections.emptyList() : plugins;